internal/venice-client-common/src/main/java/com/linkedin/venice/D2/D2ClientUtils.java

```diff
@@ -112,7 +114,11 @@ public static RestResponse sendD2GetRequest(String requestPath, D2Client client)
     try {
       response = client.restRequest(createD2GetRequest(requestPath)).get();
     } catch (Exception e) {
-      throw new VeniceException("D2 client failed to sent request, " + requestPath, e);
+      Throwable t = ExceptionUtils.getRecursiveCause(e, RestException.class);
+      if (t == null) {
+        throw new VeniceException("D2 client failed to send request, " + requestPath, e);
+      }
+      response = ((RestException) t).getResponse();
```

Comment:
It make sense to get the true root cause of the exception but why do we not throw exception now? Is `RestException` okay to not return error?

Reply:
If a D2 request fails with any reason other than a 2XX, it gets wrapped in a `RestException` ([Link](https://github.com/linkedin/rest.li/blob/master/r2-core/src/main/java/com/linkedin/r2/transport/http/common/HttpBridge.java#L70-L83)). We also return 4XX and 5XX in our codes.

In most cases, we return errors in `ErrorType` field in `ControllerResponse` along with 400 status code that we want to handle gracefully